### PR TITLE
Added endpoint '/api/articles/:article_id/comments' to post new comment with error handling

### PR DESCRIPTION
**Pull request by Ted-Smiles**
Added endpoint '/api/articles/:article_id/comments' where the endpoint returns the new comment after adding the comment to the comments table for the happy path.

Included errors for valid yet non-existent articles and invalid articles, also when the provided comment using an invalid user or invalid key returns with an appropriate message

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -153,6 +154,99 @@ describe("/api/articles/:article_id/comments",()=>{
...
+        return request(app)
+            .post("/api/articles/100/comments")
+            .send(newComment)
+            .expect(404)
+                .then(({body})=>{
+                    const {msg} = body
+                    expect(msg).toBe("article_id does not exist")
+                })
+    })
+    test("POST 404 when given an valid but non-existent article_id",()=>{
```
Comment (Ted-Smiles) [inline __tests__/app.test.js:199]: I have corrected this to a invalid article_id and a 400 err code

Comment (nc-saima-khanom) [reply __tests__/app.test.js:199]: We should also have a test for this description though - not sure if you also added that in!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -153,6 +154,99 @@ describe("/api/articles/:article_id/comments",()=>{
                     expect(msg).toBe("Invalid article_id")
                 })
     })
+
+
+
+    // Post
+    test("POST 201 and the created comment", () => {
```
Comment (nc-saima-khanom) [inline __tests__/app.test.js:161]: we could also have a test that checks any unnecessary properties are ignored

File: models/models.js [hunk 3]
```diff
@@ -52,4 +52,22 @@ exports.selectAllCommentsFromArticleId = (id) => {
             return rows
         }
     })
+}
+
+exports.updateNewComment = (newComment, id) => {
+    const { author, body } = newComment
+
+    return db.query(`
```
Comment (nc-saima-khanom) [inline models/models.js:60]: great work!

File: __tests__/app.test.js (hunk 1 above)
Comment (Ted-Smiles) [reply __tests__/app.test.js:199]: I've got a test for valid and non-existent article_id and the one above is the test is for an invalid article_id so I've changed the description now

**Pull request merged**